src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2072,6 +2077,11 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                 throw new ArgumentException(CodeAnalysisResources.MetadataPeStreamUnexpectedWhenEmittingMetadataOnly, nameof(metadataPEStream));
             }
 
+            if (metadataPEStream != null && options?.IncludePrivateMembers == true)
+            {
+                throw new ArgumentException(CodeAnalysisResources.IncludingPrivateMembersUnexpectedWhenEmittingToMetadataPeStream, nameof(metadataPEStream));
+            }
```

Comment:
As far as I can tell, we don't have a requirement for this scenario (secondary output set, yet include private members). That said, we could allow it anyways for regularity.

I listed the four scenarios and how they translate to `Emit` API in the [feature document](https://github.com/dotnet/roslyn/blob/features/refout/docs/features/refout.md).